src/main/java/cashleh/transaction/ExpenseStatement.java

```diff
@@ -85,6 +86,62 @@ public void printExpenses() {
         Ui.printMultipleText(texts);
     }
 
+    /**
+     * Finds and displays expenses that match the specified criteria, including description, amount, date, and category.
+     * @param description The description to filter transactions by. Can be left null or empty.
+     * @param amount The amount to filter transactions by. Set to -1 if no amount is provided by user.
+     * @param date The date to filter transactions by. Set to null if no date is provided by user.
+     * @param category The category to filter transactions by. Set to null if no category is provided by user
+     * @throws CashLehMissingTransactionException if no matching transactions are found.
+     */
+    public void findExpense(String description, double amount, LocalDate date, Categories category)
+            throws CashLehMissingTransactionException {
+        ArrayList<String> matchingExpenses = new ArrayList<>();
+        boolean isMatch = false;
+
+        // Customize the message based on input
+        StringBuilder message = new StringBuilder("Here are your corresponding expenses with ");
+        if (description != null && !description.isEmpty()) {
+            message.append("description: ").append(description);
+        }
+        if (amount != -1) {
+            if (description != null && !description.isEmpty()) {
+                message.append(" , ");
```

Comment:
Would it be better to just put this in your first if statement? like check for amount inside the first if statement instead of having to repeat the description check again